dask/array/ufunc.py

```diff
@@ -29,7 +34,8 @@ def wrap_elemwise(numpy_ufunc, array_wrap=False):
     def wrapped(*args, **kwargs):
         dsk = [arg for arg in args if hasattr(arg, '_elemwise')]
         if len(dsk) > 0:
-            if array_wrap:
+            if (array_wrap and
+                    (is_dataframe_like(dsk[0]) or not IS_NEP18_ACTIVE)):
```

Comment:
Wondering if it should be like this.

```suggestion
                    (is_dataframe_like(args[0]) or not IS_NEP18_ACTIVE)):
```

Reply:
(asking as we are seeing a CI failure and am curious if this would fix it)

Reply:
No, I checked this now. Testing for `is_dataframe_like` is not enough, we have also to test for `is_series_like` (and probably `is_index_like`, though I haven't seen that case yet). I can do it the same way, import all three, if importing doesn't succeed, define all three functions. Suggestions on making this cleaner? Perhaps defining another function to test if it's one of the three aforementioned?

Reply:
Ok, thanks for checking.

Importing all with fallbacks is probably fine. We could also move these to `dask/utils.py` as Matt suggested earlier and just import them without fallbacks here. The latter is sounding more appealing now that we are adding more of these 🙂

Reply:
Agreed. My latest commits takes care of that, hopefully I didn't introduce any additional issues.

Reply:
Looks good. Thanks Peter!